### PR TITLE
docker

### PR DESCRIPTION
**Pull request by ChabVlad**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+MYSQLDB_DATABASE=book_store
```
Comment (okuzan) [inline .env]: git rm --cached .env - .env should never be in version control, add .env.template instead with actual values  removed - otherwise we would need to search for all  env var throughout the project to start it

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url"  : "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+        "spring.datasource.username" : "$MYSQLDB_USER",
+        "spring.datasource.password" : "$MYSQLDB_ROOT_PASSWORD",
+        "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (okuzan) [inline docker-compose.yml]: 5005 to DEBUG_PORT

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,8 +1,6 @@
 spring.application.name=BookStore
-spring.datasource.url=jdbc:mysql://localhost/book_store?serverTimezone=UTC
+spring.datasource.url=jdbc:mysql://mysqldb:3306/book_store?serverTimezone=UTC
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]: Why did you change it?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,8 +1,6 @@
 spring.application.name=BookStore
 spring.datasource.url=jdbc:mysql://localhost/book_store?serverTimezone=UTC
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]:
```suggestion
spring.datasource.url=jdbc:mysql://localhost:3306/book_store?serverTimezone=UTC
```

Also, you can get properties from .env file `spring.config.import=optional:file:.env[.properties]`
In this case you will have like: 
spring.datasource.username=${MYSQLDB_USER} etc

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

File: .env.template [hunk 5]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=${MYSQLDB_USER}
```
Comment (okuzan) [inline .env.template]: this doesn't make much sense. It's kinda a recursion :) Better remove everything after = altogether

**Review by okuzan — CHANGES_REQUESTED** [review]

File: .env.template [hunk 6]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=user
+MYSQLDB_DATABASE=db
+MYSQLDB_ROOT_PASSWORD=pass
+MYSQLDB_LOCAL_PORT=port
+MYSQLDB_DOCKER_PORT=port
+
+SPRING_LOCAL_PORT=port
+SPRING_DOCKER_PORT=port
+DEBUG_PORT=port
```
Comment (Elena-Bruyako) [inline .env.template]:
```suggestion
MYSQLDB_ROOT_PASSWORD=
MYSQLDB_LOCAL_PORT=
MYSQLDB_DOCKER_PORT=

SPRING_LOCAL_PORT=
SPRING_DOCKER_PORT=
DEBUG_PORT=
```

File: .env.template [hunk 7]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=user
+MYSQLDB_DATABASE=db
```
Comment (Elena-Bruyako) [inline .env.template]:
```suggestion
MYSQLDB_DATABASE=
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

**Review by Elena-Bruyako — APPROVED** [review]
well done
